test/functional/repository/virtual-column/entity/User.ts

```diff
@@ -0,0 +1,27 @@
+import { Entity } from "../../../../../src/decorator/entity/Entity";
+import { Column, ColumnName, PrimaryGeneratedColumn } from "../../../../../src";
+import { concat } from "./utils";
+
+
+@Entity()
+export class User {
+    @PrimaryGeneratedColumn()
+    id: number;
+
+    @Column({
+        asVirtual: (names: ColumnName<User>, type) => {
```

Comment:
whats typeof `type`?

Reply:
DatabaseType  

```
/**
 * Database type.
 */
export type DatabaseType =
    "mysql"|
    "postgres"|
    "cockroachdb"|
    "mariadb"|
    "sqlite"|
    "cordova"|
    "react-native"|
    "nativescript" |
    "sqljs"|
    "oracle"|
    "mssql"|
    "mongodb"|
    "expo";
```

Reply:
don't think we need it. I also don't like first parameter. It might be useful but I don't like current design. I would prefer to make it parameterless for now and think what parameters we should add based on the future feedback.

Reply:
yes, may need more feedback

Reply:
I think it need  alias name of table at least, the name is no fixed.